scripts/shared/check-non-release-versions.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+tmpdir=$(mktemp -d)
+trap 'rm -rf $tmpdir' EXIT
+
+# List all submariner-io dependencies with a - in their version
+# We're looking for versions pointing to commits, of the form
+# vX.Y.Z-0.YYYYMMDDhhmmss-hash
+GOWORK=off go list -m -mod=mod -json all | \
+    jq -r 'select(.Path | contains("/submariner-io/")) | select(.Main != true) | select(.Version | contains ("-")) | select(.Version | length > 14) | "\(.Path) \(.Version)"' | \
+    while read -r project version; do
```

Comment:
Nit: Should there be indentation inside this loop?

Which highlights that [shellcheck](https://github.com/koalaman/shellcheck/issues/2463) doesn't do format checking. They recommend [shfmt](https://github.com/mvdan/sh). I can look into it if other people think it would be helpful.

Reply:
There should indeed, thanks! `shfmt` could be useful; our indentation in particular is inconsistent, running `shfmt -i 4` would fix that.

Reply:
Raised https://github.com/submariner-io/shipyard/issues/1737 to track.